utils/get_subscription.py

```diff
@@ -0,0 +1,82 @@
+from categories.models import Category,UserSubscription
+
+class get_subscription:
+    should_subscribe = {}
+    subscribe = []
+    unsubscribe = []
+    last_subscribed = {}
+    selected = {}
+    roots = []
+    
+    def __init__(self,save,drop,person,action):
+            self.save = save
+            self.drop = drop
+            self.person = person
+            self.all = save+drop
+            self.action = action
+            self.subscribe = []
+            self.unsubscribe = []
+            self.roots = []
+   
+    #nodes to be selected
+    def select(self):
+        for slug in self.save:
+            self.selected[slug] = True
+        for slug in self.drop:
+            self.selected[slug] = False
+
+    #gets all the roots in all categories
+    def get_roots(self):
```

Comment:
Would it be possible to optimise the db queries a bit, since you are already fetching all the categories here and fetching again the view separately

Reply:
I didn't directly use the returned instance as i need to store it as a key in a map and I think accessing map with model instance as a key will be of higher time complexity.

Reply:
I was able to optimize db queries but had to create a new map matching slug to model